### PR TITLE
refactor: state channel syncing logic

### PR DESCRIPTION
**Pull request by icezohu**

### CODE REVIEW

File: apps/indexer-coordinator/src/payg/payg.service.ts [hunk 1]
```diff
@@ -56,41 +61,63 @@ export class PaygService {
...
+  }
 
+  async syncChannel(channelId: string): Promise<Channel | undefined> {
+    if (!this.network.getSdk()) {
+      return;
+    }
+
+    const id = channelId.toLowerCase();
+    const channelState = await this.channelFromContract(id);
+    const channelPrice = await this.channelPriceFromContract(id);
```
Comment (ianhe8x) [inline apps/indexer-coordinator/src/payg/payg.service.ts:114]: 检查一下确保非零

File: apps/indexer-coordinator/src/payg/payg.service.ts [hunk 2]
```diff
@@ -56,41 +61,63 @@ export class PaygService {
...
     const id = channelId.toLowerCase();
-    const channel = await this.channelRepo.findOneBy({ id });
+    let channel = await this.channelRepo.findOneBy({ id });
 
-    // if (!channel) {
-    //   const channelState = await this.channelFromContract(id);
-    //   channel = await this.saveChannel(id, channelState, '0', '');
-    // }
+    if (!channel) {
+      channel = await this.syncChannel(id);
```
Comment (ianhe8x) [inline apps/indexer-coordinator/src/payg/payg.service.ts:101]: syncChannel并不能保证一定能拿到，比如传进来的是不存在的channelId，这种情况下应该返回undefined。syncChannel里好像没有处理这种情况

File: apps/indexer-coordinator/src/payg/payg.service.ts (hunk 1 above)
Comment (icezohu) [reply apps/indexer-coordinator/src/payg/payg.service.ts:114]: 保存的时候有检查，我应该还缺一步，这种情况要把Kepler network的price值存进去。

File: apps/indexer-coordinator/src/payg/payg.sync.service.ts [hunk 3]
```diff
@@ -38,79 +39,63 @@ export class PaygSyncService implements OnApplicationBootstrap {
...
       if (!hostIndexer) {
         logger.debug(`Indexer not found, will sync state channel later...`);
+        this.syncingStateChannels = false;
         return;
       }
-      const channels = await this.paygQueryService.getStateChannels(hostIndexer);
+      const stateChannels = await this.paygQueryService.getStateChannels(hostIndexer);
+      const localAliveChannels = await this.paygService.getAliveChannels();
+
+      const stateChannelIds = stateChannels.map((stateChannel) =>
```
Comment (ianhe8x) [inline apps/indexer-coordinator/src/payg/payg.sync.service.ts:58]: how about channels of status `TERMINATED`, but is `OPEN` in local db? Should call syncChannel for them as well

File: apps/indexer-coordinator/src/payg/payg.service.ts (hunk 2 above)
Comment (icezohu) [reply apps/indexer-coordinator/src/payg/payg.service.ts:101]: 这种情况下会返回undefined，可以更显式判断一下

File: apps/indexer-coordinator/src/payg/payg.sync.service.ts [hunk 4]
```diff
@@ -169,11 +154,8 @@ export class PaygSyncService implements OnApplicationBootstrap {
   }
 
   async syncOpen(id: string, consumer: string, agent: string, price: string) {
-    let channel = await this.paygService.channel(id);
-    if (!channel) {
-      const channelState = await this.paygService.channelFromContract(id);
-      channel = await this.paygService.saveChannel(id, channelState, price, agent);
-    }
+    const channel = await this.paygService.channel(id);
```
Comment (ianhe8x) [inline apps/indexer-coordinator/src/payg/payg.sync.service.ts:157]: syncOpen is only called from the event listener. We can trust the input so no need to retrieve details again from contract.

Comment (ianhe8x) [inline apps/indexer-coordinator/src/payg/payg.sync.service.ts:157]:
```suggestion
    const channel = await this.paygService.saveChannel(id, channelState, price, agent);
```

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**